Add kuzzle-common-objects and kuzzle-plugin-context types

dbengsch opened the pull request:
To be reviewed

scottinet commented on types/kuzzle-common-objects/index.d.ts:
```diff
@@ -0,0 +1,204 @@
+    toJSON(): {
+        message:string
+        status: number
+        stack: string
```
Will soon be restricted to development mode (see https://github.com/kuzzleio/kuzzle/issues/747), so I don't know if it's wise to describe this property.

scottinet commented on types/kuzzle-common-objects/index.d.ts:
```diff
@@ -0,0 +1,204 @@
+    toJSON(): {
+        message:string
+        status: number
+        stack: string
```
ditto

beaulac committed:
Add @google-cloud/datastore definitions

alloy committed:
[react-tracking] Add support for v4.2

googol committed:
Type the named export Big of big.js

googol committed:
Remove global big.js declaration

googol committed:
Change tests to reflect the new typings

marvinhagemeister committed:
Add typings for puppeteer

googol committed:
Add linting

valeriangalliat committed:
docopt: fix export format

googol committed:
Add global definition block

Ailrun committed:
Update openlayers layer options

googol committed:
Enable strict null checking

uniqueiniquity committed:
Merge pull request #19893 from bayssmekanique/master

Adds constructor with options to chokidar

uniqueiniquity committed:
Merge pull request #19824 from pasieronen/bunyan-reemit-error-events

bunyan: add support for reemitErrorEvents stream flag

uniqueiniquity committed:
Merge pull request #19807 from li-xinyang/patch-1

Add missing Data attribute for Treemap component

scleriot committed:
Update Player with new function requestFullscreen/cancelFullscreen

But keep the legacy functions

scleriot committed:
Update Videojs version

scleriot committed:
Replace cancelFullScreen by exitFullscreen

nomaed committed:
Fix for JQueryUI ResizableEvents interface (#19904)

Fixes #19901

Commit:
Added more specific typings for some interfaces

scleriot committed:
Fix space in comments

firsttris committed:
jest: added restoreAllMocks

firsttris committed:
jest: added restoreAllMocks

dudeofawesome committed:
fix(read-pkg-up): make requested changes

dudeofawesome committed:
test(read-pkg-up): adds more complete tests

uniqueiniquity committed:
Merge pull request #19803 from danielearwicker/react-foundation

Adding definitions for react-foundation

uniqueiniquity committed:
Merge pull request #19892 from bfeltman/master

Make all push-options optional

dudeofawesome committed:
style(read-pkg-up): fix linting

pettyalex committed:
Correctly mark viewBox, xAxis, yAxis as optional and add some more shapes to the test

uniqueiniquity committed:
Merge pull request #19884 from cameron-mcateer/master

Add types for react-form@1.3

uniqueiniquity committed:
Merge pull request #19792 from scottharwell/express-brute-redis

Add typings for express-brute-redis

uniqueiniquity committed:
Merge pull request #19827 from marvinhagemeister/puppeteer

Add typings for `puppeteer` (headless chrome framework)

DenisCarriere committed:
Access to Types from namespace OSRM

DenisCarriere committed:
Fix linting issue

Commit:
Add definitions for deep-freeze-es6

uniqueiniquity committed:
Merge pull request #19754 from YunS-Stacy/master

[leaflet-draw] Add Draw Handlers

maniacalmaniac committed:
Added react-native-safari-view types

maniacalmaniac committed:
tsconfig update

maniacalmaniac committed:
Used dts-gen to simplify things

maniacalmaniac committed:
fixed lint errors

firsttris committed:
bump jest version to 21.1

beaulac committed:
@google-cloud/datastore: Remove unnecessary generics.

beaulac committed:
@google-cloud/datastore: Add suppressions for unsatisfiable no-duplicate-import lint rule.

pasieronen committed:
Add types for bunyan-bugsnag

Commit:
[Puppeteer] Launch options are now optional

You can now pass the launch options 1 by 1 instead of having to pass all
of them.

Commit:
Fixed pull request build #44258 error

Commit:
Merge branch 'master' into master

Commit:
Fixed travis build #45656.1 error

Commit:
Merge branch 'master' of github.com:nishantkyal/DefinitelyTyped

keton committed:
Fix of: More general overload should come last

Commit:
Fixed spacing in comment

Commit:
[accept-language-parser] Add accept-language-parser

This commit adds initial support for the accept-language-parser module which is currently at version 1.4.1

I am not in any way related to the module.

Commit:
[accept-language-header] Confirm to dtslint

Loghorn committed:
Fixes issues with callback parameters and adds missing promisified

Commit:
[accept-language-parser] Fix return type of pick() and add tests for return types

Commit:
angular-file-upload: requested definition Fix #5322

Commit:
angular-file-upload: add tslint.json

Commit:
angular-file-upload: fix build failure on must specify lib

Commit:
angular-file-upload: add typescript version information

Commit:
angular-file-upload: 2.5 doesn't parse, let's agree on 2.1

Commit:
angular-file-upload: fixing all linting errors

Commit:
Made more options optional

Commit:
angular-file-upload: add FormData and some comments

Commit:
angular-file-upload: add defaults, object and Partial

Commit:
angular-file-upload: fix defaults for Filters

jpap committed:
Add missing tls.connect lookup option for node 8.x (#19861)

* Add missing tls.connect lookup option for node 8.x

* Following @Flarna's review: updated options type

Commit:
Change Selection.content from Slice to a function returning a Slice

Loghorn committed:
Fix whatwg-streams

Loghorn committed:
Fix optics-agent

Loghorn committed:
Add execFile tests

wy193777 committed:
fix some missing semicolon

Commit:
[moment-timezone] Add Zone population field

Population information was added in moment-timezone 0.5.0:
https://github.com/moment/moment-timezone/commit/dab50e8bcbf2d98ae92e5c688f6849c830a0a791

Zone objects have a population field as seen here:
https://github.com/moment/moment-timezone/blob/0.5.0/moment-timezone.js#L149

Its type is a number. If a timezone doesn't have a population, it defaults to 0, as seen here:
https://github.com/moment/moment-timezone/blob/0.5.0/moment-timezone.js#L129

dannycochran committed:
define ImageRequireSource instead of using number

uniqueiniquity committed:
Merge pull request #19883 from dudeofawesome/read-pkg-up

Add typings for read-pkg-up

uniqueiniquity committed:
Merge pull request #19885 from dudeofawesome/gulp-zip

Add typings for gulp-zip

uniqueiniquity committed:
Merge pull request #19881 from DenisCarriere/osrm

Publish osrm Typescript definition

uniqueiniquity committed:
Merge pull request #19757 from keton/noble-fixes

[noble] add missing noble.removeListener() interface and noble.state variable

uniqueiniquity committed:
Merge pull request #19917 from pettyalex/master

Fix types for Recharts, optional properties were incorrectly marked required

uniqueiniquity committed:
Merge pull request #19903 from ksaifullah/patch-1

Missing options for @types/mongoose

uniqueiniquity committed:
Merge pull request #19915 from firsttris/master

jest: restoreAllMocks()

uniqueiniquity committed:
Merge pull request #19913 from scleriot/master

Update Video.js Player with new function requestFullscreen/cancelFullscreen

uniqueiniquity committed:
Merge pull request #19944 from wy193777/master

[mongoose] fix some missing semicolon

uniqueiniquity committed:
Merge pull request #19771 from axetroy/master

[sequelize] describe the transaction's options as clearly as possible

uniqueiniquity committed:
Merge pull request #19778 from harrygr/patch-1

Update parameter for token callback

uniqueiniquity committed:
Merge pull request #19797 from nikeee/patch-2

Add missing cookies property

uniqueiniquity committed:
Merge pull request #19781 from forabi/update-jszip

Update JSZip

uniqueiniquity committed:
Merge pull request #19820 from jeffwu85182/master

Add props for BarcodeScanOptions

uniqueiniquity committed:
Merge pull request #19890 from dan-j/react-stripe-elements

Added definitions for 'react-stripe-elements'

uniqueiniquity committed:
Merge pull request #19919 from mattbishop/deep-freeze-es6

Add definitions for deep-freeze-es6

uniqueiniquity committed:
Merge pull request #19923 from mrand01/react-native-safari-view

Add definitions for react-native-safari-view

uniqueiniquity committed:
Merge pull request #19922 from beaulac/google-cloud-datastore

Add @google-cloud/datastore definitions

uniqueiniquity committed:
Merge pull request #19929 from Kampfgnom/add_accept_language_parser

[accept-language-parser] Add new types for accept-language-parser

uniqueiniquity committed:
Merge pull request #19932 from cyrilgandon/angular-file-upload

Add definitions for angular-file-upload

Commit:
done-callback supports a optional error parameter

pasieronen committed:
bunyan-bugsnag: expose Options interface

Loghorn committed:
Refactoring of optics-agent

kondi committed:
Add definitions for 'bricks.js'

uniqueiniquity committed:
Merge pull request #19925 from pasieronen/bunyan-bugsnag

bunyan-bugsnag: add types

uniqueiniquity committed:
Merge pull request #19926 from Robinfr/puppeteer-optional-options

[Puppeteer] Options are now optional

uniqueiniquity committed:
Merge pull request #19891 from Ailrun/openlayers

Update openlayers layer options

uniqueiniquity committed:
Merge pull request #19389 from nishantkyal/master

Added support for nightmare-inline-download to nightmare.js

uniqueiniquity committed:
Merge pull request #19947 from brieb/moment-timezone-population

[moment-timezone] Add Zone population field

uniqueiniquity committed:
Merge pull request #19931 from Loghorn/execfile

Fixes issues with callback parameters and adds missing promisified

uniqueiniquity committed:
Merge pull request #19964 from kondi/bricks.js

Add definitions for 'bricks.js'

uniqueiniquity committed:
Merge pull request #19801 from KillerCodeMonkey/master

@types/quill - add strict and scrollContainer fields to QuillOptionsStatic interface

dannycochran committed:
remove whitespace

uniqueiniquity committed:
Merge pull request #19954 from ckanibal/mongoose-fix

@types/mongoose: done-callback supports a optional error parameter

uniqueiniquity committed:
Merge pull request #19939 from davidka/change-wrong-selection-content-type

Change wrong selection content type

uniqueiniquity committed:
Merge pull request #19813 from matrumz/master

[better-sqlite3] Added fileMustExist to DatabaseOptions

lavyun committed:
update module import

alloy committed:
Merge pull request #19047 from dannycochran/patch-8

update react-native image source typing

spellman committed:
[ramda]: Correct order of reducing-fn params in reduceRight (#18798)

The reducing function in Ramda's reduceRight has signature
(elem: T, acc: TResult) => TResult,
as per https://github.com/ramda/ramda/blob/v0.24.0/src/reduceRight.js.

The order of the parameters was switched at
https://github.com/ramda/ramda/commit/68d1dc19fb035b0fcafcc1857331f9d00986470f

Stamo-Gochev committed:
feat: upgrade Kendo UI definitions to 2017.3.913

The fixes for Alert, Confirm and Prompt messages in:

https://github.com/DefinitelyTyped/DefinitelyTyped/commit/332ca267d198a271683921ea750ea4cf15bcdf57#diff-6892c3337024d80c66687f1e88553bf2

are no longer necessary as they are addressed by us in:

https://github.com/telerik/kendo/commit/23ea31e4fbef78f292295dea7c2d739240a7a114#diff-8192a644b53dbe67e6aeddfe95e4d12a

googol committed:
Add separate tests for the global exports

nycdotnet committed:
Merge pull request #19829 from googol/bigjs

[big.js] named module export

valeriangalliat committed:
docopt: export DocoptOption

uniqueiniquity committed:
Merge pull request #19805 from lavyun/master

Add declarations for 'mockjs'

uniqueiniquity committed:
Merge pull request #19782 from dobrakmato/patch-3

Correct interface fields in FfprobeData

uniqueiniquity committed:
Merge pull request #19739 from telerik/kendo-ui-2017.3.913

feat: upgrade Kendo UI definitions to 2017.3.913

uniqueiniquity committed:
Merge pull request #19855 from valeriangalliat/docopt-export

Fix docopt export format

smhxx committed:
Adding CompositeDisposable class and finishing IConfig methods (#19993)

Commit:
@types/echarts Modify namespaces from ECharts to echarts (#19507)

* Modify namespaces from ECharts to echarts

Modify namespaces from ECharts to echarts. Because the global variables in the window browser using echarts rather than ECharts.
If I don't use UMD or AMD or commonjs way to use echarts, ECharts compiled in the browser can not find.

* ECharts is only a type not a constructed function

plantain-00 committed:
[React] change createElement declaration (#19858)

* fix https://github.com/Microsoft/TypeScript/issues/15019

* change the stable v15 declaration too

* catch-all case also take string

* add string for test

SheetJSDev committed:
remove `xlsx` types (#19889)

original discussion:
- DefinitelyTyped/DefinitelyTyped#16500
- SheetJS/js-xlsx#655

uniqueiniquity committed:
Merge pull request #19768 from amritk/master

Added some missing members to googlemaps opening_hours

royxue committed:
[react-datepicker]Add select time feature for react-datepicker 0.55 (#20009)

* Add select time feature for react-datepicker

* Fix lint error

chrislgrigg committed:
ChartLegendLabelOptions.fontStyle is a string (#20006)

Per docs at http://www.chartjs.org/docs/latest/configuration/legend.html#legend-label-configuration. Tested in code, string works correctly.

codeaid committed:
Update react-dropzone definitions to v4.1.3 (#20004)

* Moved type definitions for v3.13 to a subdirectory

* Updated react-dropzone type definitions to v4.1.3

ganlhi committed:
Removing myself from authors (#19971)

* [openlayers] Added missing functions and configuration property added by openlayers 4.1.0

* Added tests

* Removed myself from authors

Den-dp committed:
Handy imports for angular-toastr (#19967)

This change to the module definition allows ES6 style imports, e.g. import { IToastBaseConfig } from 'angular-toastr';

djcsdy committed:
[restify] Correct and tighten type declarations. (#19909)

* Remove spurious exports.

restify does not export the constructors for Server or
Router.

* Correct type of 'bunyan' export.

I'm not sure where the previous definitions came from, but
they definitely do not match restify 5.0.1.

restify.bunyan is not callable. It is a namespace that
contains various helper functions and objects relating to
logging with bunyan.

* Specify stricter types for ServerOptions.

`ca`, `certificate` and `key` are passed through to as
the corresponding options to https.createServer().

`spdy` is passed through to spdy.createServer().

* Correct type signature for formatters.

Psvensso committed:
Default export and removing namespace (#19863)

* Default export

* Correct imports for tests

* Update react-virtualized-select-tests.tsx

* Correct imports and cleaning

Djemel committed:
@types/webpack - output.library should allow both string and string[] (#19959)

output.library allows for string array like this
    output: {
        filename: './dist/dist.js',
        library: ['namespaceA', 'solutionA']
    }

z0al committed:
(webpack): add missing possible 'entry' types (#19977)

* fix(webpack): add missing possible 'entry' types

resolves #19972

* style(webpack): fix spacing

* style(webpack): add parentheses

I've added parentheses around the return type of `EntryFunc` to clarify
the correct type is `() => (...types)` not `(() => string) | ...types`

calebboyd committed:
fix: add static named Promise export (#19859)

jsheely committed:
mapValues can accept a single param and output entire object (#19786)

Thecentury committed:
Added definitions for dateDisabled and cuscomClass options. (#19694)

kylevogt committed:
Heroku logging (#19969)

* Created typings for heroku-logger

* add support for constructor function that is also exported by heroku-logger

Commit:
[axios-mock-adapter] add typings (#19973)

* [axios-mock-adapter] add typings

* [axios-mock-adapter] include dependencies and fix linting issues

zspitz committed:
Microsoft Office (Desktop) via Automation (#19987)

* Merge updates (#4)

* Merge upstream updates (#5)

* Initial commit for MS OFfice Automation

* Modified activex-excel and tests; activex-msforms

* ADODB tests, DAO tests

* msforms, msxml, mshtml

* MSXML tests

* Office, VBIDE, Word

* Word

* Word

* Outlook, Powerpoint

* Added project URL for stdole

* Added 'private' to package.json

* activex-adodb fixes

* remove public member-access

* activex-adodb fixes

* activex-adodb tests fixes

* activex-adodb-tests fix

* activex-adodb-tests fix -- no-unnecessary-generics

* remove public modifier

* TravisCI fixes

* More TravisCI fixes

* More TravisCI fixes

DanielRosenwasser committed:
Added type declarations for 'documentdb-session'. (#19992)

pasieronen committed:
passport-oauth2: add types (#20012)

voxmatt committed:
minor: add "playsinline" attribute to <video/> (#19530)

* adding optional 'playsinline' attribute to HTML5 video tag

* also adding playsinline to v15

efokschaner committed:
three: BufferGeometry.setIndex accepts number[] (#19897)

This PR includes the suggestions recommended (by myself) in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19577.
If we merge this, please close the other.
Thanks @assafcw for identifying the original fix.

discharged-spider committed:
Fixed Navigator.renderPage. (#19728)

Added new parameter to Navigator.renderPage, because Navigator does not works without it.

e-cloud committed:
Types/lodash -- update with new functions (#18738)

* feat(lodash): update types and list of lodash functions

* fix: add missing typings for flatMapDepth

* test(lodash): update test with newly added function

Note: pullAllWith is not included

* fix: add missing entries in tsconfig.json

* feat: add and generate new modules

* fix(lodash): improve typings

* fix: update with latest flatMap types

* fix(lodash): remove redundant types for updateWith method

* fix(lodash): improve types of flatMapDeep and flatMapDepth and tests

replace Many with ListOfRecursiveArraysOrValues

guybedford committed:
Fix up URL iteration (#19683)

* fix up URL interfaces

* try removing Iterable implements

* es5 support fix

natecj committed:
Increase allowed mixin params to Ember.Component.extend (#19594)

Commit:
add type for leaflet-gpx (#20015)

WorldMaker committed:
Add image quality options to blob-util (#20014)

* Add image quality options to blob-util

* Remove unapplicable doc comment note from blob-util

j-f1 committed:
Add typings for to-camel-case (#20013)

outring committed:
Updated to version 1.3.4 (#19985)

doshitan committed:
[pg] Client.connect() returns void not a Client (#19918)

https://node-postgres.com/api/client#-code-client-connect-gt-promise-lt-void-gt-code-

DenisCarriere committed:
Set default export to Polylabel (#19869)

bergi9 committed:
[React-Router-DOM] Fix exporting types for JetBrains IDE (#19856)

* Fix exporting types for JetBrains IDE

Rider always suggest me to use require instead of import because it didn't detect the imported type in the code.
Since this definitions use the declare module and the react-router definitions don't.

* Fix testing for trailing whitespace

gavdraper committed:
Update definitions to support react-big-calendar Event object properties for OnSelectEvent (Version 0.15.0) (#19832)

* Update definitions to support react-big-calendar Event object properties

* Update version number

jfloff committed:
[PapaParse] Comments if set are a string with the comment char. (#19852)

* Comments if set are a string with the comment char

See more: http://papaparse.com/docs#config

* Add myself to definitions authors

Commit:
[adone] refactoring, new typings (#19812)

* [adone] refactoring, new typings
* refactor to namespace style
* move adone.run to adone.application
* [std] remove unnecessary std.ts code
* move is and x to separate dirs
* init collections, add LinkedList typings, move ExBuffer here
* init event, move EventEmitter and AsyncEmitter here
* init fs
* init fast
* init streams, add core typings
* remove common
* [util] add userdir, LogRotator, debounce, uuid.v3, fix entries
* [shani] add spy.waitFor* definitions
* [is] add descriptions
* [promise] callbackify

* [adone] rename is.exbuffer to is.byteArray

* [adone] add data typings

* [adone] rename is.exdate to is.datetime

* [adone] fs fixes

* [adone] add datetime typings

* [adone] improve datetime help messages

* [adone] math
* decomposition
* add descriptions for Long
* add typings for matrix, simd, BitSet, BigNumber, random

* collections, math
* add some lazify options
* [collections] add help messages for ByteArray
* [collections] add typings for Queue, AsyncQueue, PriorityQueue, FastLRU, Stack, BinarySearchTree, AVLTree, RedBlackTree, ArraySet, BufferList, DefaultMap, LRU, TimedoutMap
* [math] add missing BitSet methods, normalize help messages

* [adone] compressors, fast
* [fast] change FIle.is* methods
* [compressors] init

* [adone] archives, fs, datetime
* [archives] init, tar, zip
* [fs] add AbstractRandomAccessReader, RandomAccessFdReader, RandomAccessBufferReader
* [datetime] add support for ms-dos format

* is, streams, fast
* [fast] fix typo
* [is] fix fast stream predicates, add some runtime predicates
* [streams] more detailed core help messages

danielrbradley committed:
Fix parameter name of toBeCloseTo (#19826)

The second argument passed in is not the amount that the number should be within, but is actually the number of significant digits to check. 

Reference: https://github.com/facebook/jest/blob/master/docs/en/ExpectAPI.md#tobeclosetonumber-numdigits

FishOrBear committed:
Clean code (#19818)

nrlquaker committed:
Update electron-json-storage to the latest version (#19810)

kreba committed:
@types/reactstrap Add prop autoFocus for Modal component (#19791)

* @types/reactstrap Add prop autoFocus for Modal component

* bump version number to 4.6

royxue committed:
[recharts] Improve Label props (#20016)

* Fix typo and is required

* Add new viewbox type

* Change tp correct contributor github link

davidka committed:
Update prosemirror-model to v0.24 (#19935)

* Make NodeRange constructor public

* Remove parameter from ResolvePos.marks

* Remove matchNode, matchToEnd, findWrappingFor and allowsMark from ContentMatch

* Add allowsMarkType, allowsMarks and allowedMarks to NodeType

* Change ContentMatch.validEnd from method to a property

* Change return type of ContentMatch.findWrapping

* Remove AttributeSpec.compute

* Remove Fragment.offsetAt

* DOMParser.schemaRules is no longer public

* Replace topStart from DOM parser option with topMatch

* DOMSerializer methods nodesFromSchema and marksFromSchema are no longer public

* ParseRule.contentElement may now be a function.

* Add new method ResolvedPos.marksAcross

* Add NodeType.contentMatch

* Add NodeSpec.marks

* Add Fragment.size

* Increment version for prosemirror-model to 0.24

* Adapt parameter and return type of ContentMatch.findWrapping

avchugaev committed:
[prop-types] Added missing definition of PropTypes.symbol (#19970)

* Added definition for PropTypes.symbol

* Tests of support PropTypes.symbol

Commit:
[redux-logger] Allowing color configuration to be false (#19978)

* Allowing color configuration to be false

* Added test for overall color boolean

bolatovumar committed:
[jasmine-enzyme] Add default export (#19991)

* Add type definitions for jasmine-enzyme

* [jasmine-enzyme] Add default export

stuartjo-sportsbet committed:
[react-router-config] - add support for optional extraProps argument to renderRoutes introduced in react-router 4.2.0 (#19994)

davidka committed:
Update prosemirror transform state and view to v0.24 (#20002)

* Move offset method from Step to StepMap

* Replace clearMarkup inside of Transform with clearIncompatible

* Increment version for prosemirror-transform to 0.24

* Adapt prosemirror-state tests to fit prosemirror-transform v0.23

* Increment version for prosemirror-state to 0.24

* Replace Transform.setNodeType with Transform.setNodeMarkup

* Add domAtPos method to EditorView

* Remove onFocus, onBlur, and handleContextMenu from EditorView

* Add EditorProps.handleScrollToSelection

* Add clipboardTextParser and clipboardTextSerializer to EditorProps

* Extend DecorationSet.find with parameter predicate

* Increment prosemirror-view version to v0.24

* Adapt prosemirror-state tests to fit v0.24 of prosemirror-transform

plantain-00 committed:
add options for get-port (#19982)

justinmwarner committed:
Add contextMenu event to Simulate on TestUtils (#19854)

* Add contextMenu event to Simulate on TestUtils

* Reverting format on save

natesilva committed:
declare the namespace (#20008)

allows `import * as pLimit from 'p-limit'`

jayongg committed:
Catch up Microsoft Teams SDK with what we support today. (#20019)

* Added some new types/methods.

context.isFullScreen?
microsoftteams.getTabInstances

as well as the TabInstance and TabInformation interfaces

* add the teamType type

* Update to remove unnecessary/incorrect comment.

* Update Microsoft Teams's index.d.ts to match current spec.

index.d.ts modified by copying the output of the microsoft-teams-library-js resultant .d.ts file

* Increase the version number in the header of the index.d.ts

* Fixed tslint error

RyanCavanaugh committed:
Correct spelling of parsers (#20018)

* Correct spelling of parsers

```
C:\Throwaway\spp>node
> require('serialport').parsers
{ ByteLength: [Function: ByteLengthParser],
  Delimiter: [Function: DelimiterParser],
  Readline: [Function: ReadLineParser],
  Ready: [Function: ReadyParser],
  Regex: [Function: RegexParser] }
```

* Fix tests

evanshortiss committed:
Add "tocktimer" typings (#20017)

* add types for tocktimer

* use correct formatting

* add linting and update structure

* update for build issues

acchou committed:
Add typing for Recompose `createEventHandlerWithConfig()`, which will be exported. See https://github.com/acdlite/recompose/pull/522 (#19921)

RomkeVdMeulen committed:
cucumber: define overridable World interface (#19867)

* cucumber: define overridable World interface

By implementing World as an interface, it can be extended in projects by using declaration merging.

* cucumber: add example World override to test

tyronedougherty committed:
Allow @types/angular-loading-bar to export a string name for angular module inclusion (#19888)

alloy committed:
Merge pull request #20011 from alloy/react-tracking

Add typings for react-tracking

Commit:
async: Convert to module (#19870)

Commit:
backbone.marionette: Fix lint (#19846)

Commit:
Prefer relative imports for packages that import theirselves (#20034)

kgtkr committed:
add react-youtube definition file (#20037)

* add react-youtube

* strictNullChecks:true

cyrilgandon committed:
Add new typings for cordova-plugin-file-opener2 (#20027)

Commit:
@types/dwt: Add Load and AutoLoad to WebTwainEnv to allow for manual loading (#20007)

* Add Load and AutoLoad to WebTwainEnv

* Update versions in header

* Revert typescript version back to 2.2

* Add v12 folder

Commit:
Fixes definitions to match reality. (#19800)

Return type on ignore function was missing and Promise return type was missing from overloads that didn't take a callback.  Also extracts out a helper type definitions and provides slightly better variable naming for the transient `recursiveReadDir`.

supermacro committed:
Make client constructor options param optional (#20041)

supercargo committed:
[vis] timeline: add types for custom ordering functions (#20031)

* add types for custom ordering functions

* fix trailing whitespace

* include dependency on moment  (2.13 being the first version to include types)
add type for moment constructor for use in timeline and graph2d options

davidka committed:
Increment prosemirror-keymap version to 0.24 (#20024)

davidka committed:
Increment version of prosemirror-command to v0.24 (#20023)

levibotelho committed:
Add Feather icons to react-native-vector-icons (#19995)

* Add Feather icons to react-native-vector-icons

* Bump react-native-vector-icons version number

* Add Feather.d.ts to react-native-vector-icons tsconfig

* Rename Feather.ts to Feather.d.ts in react-native-vector-icons

florian-richter committed:
Revert and fix lint/tests (#19957)

Commit:
lambda-phi: Provides its own types (#20035)

Commit:
request-promise-native: Convert to external module style (#20036)

Commit:
cloudflare-apps: Remove unnecessary `declare module` (#20043)

Commit:
lunr: Use `export as namespace` instead of global declarations + `declare module` (#20045)

iwllyu committed:
add isPostalCode (#20042)

nycdotnet committed:
Document how to write definitions for modules that can also be used globally (#20033)

* document writing definitions for global and module libraries

* minor edits

jzoric committed:
fix(react-sortable-tree): Add missing prop (#20029)

Add missing dndType prop.

TakenakaSimon committed:
fix(twit): add some missing params (#20026)

Commit:
Add tslint disables for "no-declare-current-package" (#20044)

tony19 committed:
Fix node-spdy createServer signature (#20022)

* Fix spdy createServer signature

* Fix spdy lint errors

Commit:
react-intl: Add ignore for "no-declare-current-package" (#20048)

Commit:
jasmine-enzyme: Convert to external module (#20047)

Commit:
next: Convert to external modules (#20046)

ryota-ka committed:
[chrome] Add and utilize ColorArray type (#19836)

* Add and utilize ColorArray type

* fixup! Add and utilize ColorArray type

Commit:
graceful-fs: Convert to external module (#20049)

morphatic committed:
Updating existing @types/d3kit definition to new version: 3.1.2 ==> 3.2 (#19952)

* update @types/d3kit to 3.2

* update @types/d3kit to 3.2

cbargren committed:
Fix type definitions for Hapi load configuration property (#19949)

Any or all of the properties in the load configuration object can be defined.

ckotzbauer committed:
updated extended-listbox to 4.0.0 (#19943)

snaik2064 committed:
visible property added in IndividualSeriesOptions interface. (#19955)

kant2002 committed:
[react-bootstrap-table] Add thStyle, tdStyle, tdAttr properties to TableHeaderColumn (#19910)

* Add thStyle, tdStyle, tdAttr properties to TableHeaderColumn
Also style apply TSLint rules where it was fast to implement without introducing issues.

* Fix missed TSLint warnings

bodeyb committed:
Adds missing scrollbarStyle EditorConfiguration attribute (#20050)

[Codemirror documentation](http://codemirror.net/doc/manual.html) (search for `scrollbarStyle`)

Loghorn committed:
Add missing node-ipc configuration options (#19942)

* Add missing options

* Use method signature

Commit:
Add null as a type for bsStyle in ButtonProps to allow for custom classes (#19941)

fabio-paiva-sp committed:
Add new properties from Bootstrap Beta to Navbar (#19940)

Reactstrap Navbar new properties tests

Add contributor to be notified on changes

NaridaL committed:
[mz] Added fs.mkdtemp (#19928)

screendriver committed:
[loglevel] Refactored and added log.levels property (#19742)

* Refactored loglevel

* Fixed linting errors

* Moved tests to dedicated test directory

mad-mike committed:
[pify] Update to version 3.0.0 (#20038)

* [pify] Add missing option errorFirst

see https://github.com/sindresorhus/pify#errorfirst

* [pify] Migrate promiseModule

* [pify] Update definitions to version 3.0.0

kevin-greene-ck committed:
[node-int64] Updating for missing methods, props and constructors (#20053)

* Supports constructing from number
  * Support the public buffer and offset properties
  * Support all method signatures of the setValue method

vilterp committed:
Fix dagre types to allow objects to be set as default edge labels (#19945)

The tests call `.setDefaultEdgeLabel(() => {})` which is a translation
to ES6 of something seen in the example in [Dagre's
readme](https://github.com/cpettitt/dagre/wiki#an-example-layout):
`g.setDefaultEdgeLabel(function() { return {}; });`. However, the
translation is botched: `() => {}` doesn't return an empty object, it
returns `undefined`, because `{}` is not an object literal, it's a
block with 0 statements.

I think this is what motivated the current typing to specify that `void`
can be returned by the callback. However, in my experience returning
`undefined` from this callback causes the library to crash. Thus I've
chagned the tests to say `.setDefaultEdgeLabel(() => ({}))` (i.e.
actually return an empty object, which is what the author probably
intended), and updated the types accordingly. This works in my project.

vaskevich committed:
Add missing ReactDOM Simulate event methods (#19980)

* Add missing ReactDOM Simulate event methods

Simulate has a method for every event that React understands.

Source: https://facebook.github.io/react/docs/test-utils.html#simulate

* Fix ordering

mgroenhoff committed:
Allow to assign result of Sequelize.where() to where clause in find options. (#19962)

DenisCarriere committed:
Allow access to custom GeoJSON properties @types/geojson (#19845)

* Allow access to custom GeoJSON properties @types/geojson

* Specify Typescript Version 2.2

* GeoJSON replace properties with "any"

Jazcash committed:
Added objectMode option (#19360)

dkamburov committed:
Update Ignite UI typings, adding some missing event properties (#19934)

lordazzi committed:
Type definition for Lazar Laszlo's jsqrcode lib (#20051)

* Mapeando a estrutura mais simples da biblioteca e criando um tsconfig.json para as definições

* Unindo todas as definições em um único arquivo e adicionando alguma documentação

* Criando modulo fantasma

* Adicionando autoria

* Removing all any type, creating tsconfig.json and tslint.json

* Reviewing some of the warnings of tslint

* solving the problems indicated by tslint

* Setting 'strictNullChecks' flag as true

elton2048 committed:
Change the way to express type.
Fill all methods correspond to the library.

tomwanzek committed:
d3: (#20059)

* [FIX] Add missing `version` property

Commit:
plotly.js: Fix no-self-import (#20075)

uniqueiniquity committed:
Merge pull request #19830 from elton2048/soundmanager2

Add definitions for soundmanager2

zero51 committed:
Add declarations for 'countries-and-timezones' (#20055)

lordazzi committed:
Types for cordova CanvasCamera plugin (#20061)

Commit:
Add declarations for koa-range 0.3 (#20071)

varHarrie committed:
Add declarations for 'react-beautiful-dnd' (#20060)

* Add declarations for 'react-beautiful-dnd'

* Add missing header

* Fix lint issues

* Fix mistake

* Remove personal tslint rules

Commit:
Add declarations for end-of-stream 1.4 (#20076)

kgtkr committed:
bug fix:"File '/node_modules/@types/kuromoji/index.d.ts' is not a module." (#20066)

add tslint.json

Commit:
express-flash-2: Convert to external module (#20081)

Commit:
popper.js: Fix lint (#20084)

Commit:
iframe-resizer: Convert to external module (#20082)

estaub committed:
Add onClick to Avatar, Chip (#20079)

* Add onClick to Avatar

This superficially addresses https://github.com/DefinitelyTyped/DefinitelyTyped/issues/19988, which should probably be closed when this is pulled, for lack of interest.

* Added onClick to ChipProps

tsvetomir committed:
fix: allow custom filters (#20070)

simontreny committed:
Add missing extractOffset() methods and Animated.ScrollView

alloy committed:
Merge pull request #20086 from simontreny/master

[react-native] Add missing extractOffset() methods and Animated.ScrollView

Commit:
mongoose-simple-random: Convert to external module (#20083)

mchaov committed:
Updated Enzyme typings (#20073)

screendriver committed:
[loglevel] Moved everything to a namespace called log (#20063)

HenriqueCaires committed:
Add timeout option (#19951)

Maistho committed:
Update Navbar.d.ts (#19878)

navExpanded was renamed to expanded in https://github.com/react-bootstrap/react-bootstrap/commit/f9ea411

wy193777 committed:
fix some semicolon issues for elasticsearch (#19965)

dbengsch committed:
Merge remote-tracking branch 'gh/master' into topic-kuzzle-plugin
